src/uu/split/src/split.rs

```diff
@@ -411,15 +435,48 @@ impl Strategy {
 
 /// Parse the suffix type from the command-line arguments.
 fn suffix_type_from(matches: &ArgMatches) -> Result<(SuffixType, usize), SettingsError> {
-    if matches.value_source(OPT_NUMERIC_SUFFIXES) == Some(ValueSource::CommandLine) {
-        let suffix_start = matches.get_one::<String>(OPT_NUMERIC_SUFFIXES);
+    // Collect provided suffixes (if any)
+    // Any combination is allowed
+    let mut suffixes = vec![];
+    for suffix_id in [
+        OPT_NUMERIC_SUFFIXES,
+        OPT_NUMERIC_SUFFIXES_SHORT,
+        OPT_HEX_SUFFIXES,
+        OPT_HEX_SUFFIXES_SHORT,
+    ]
+    .into_iter()
+    {
+        if matches.value_source(suffix_id) == Some(ValueSource::CommandLine) {
+            suffixes.push((matches.index_of(suffix_id), suffix_id));
+        }
+    }
+    // Compare suffixes by the order specified in command line
+    // last one (highest index) is effective, all others are ignored
```

Comment:
It's much easier to make all the arguments override each other :)

Reply:
yep, refactoring with `overrides_with_all()` in args definitions